experimental/LieAlgebras/src/root_systems.jl

```diff
@@ -0,0 +1,226 @@
+#
+###############################################################################
+@doc raw"""
+  number_of_roots(R::RootSystem)
+
+Return the numbers of roots in the root system `R`
+"""
+number_of_roots(R::RootSystem) = size(R.roots)[1]
+
+@doc raw"""
+  number_of_roots(S::String)
+
+Return the numbers of roots in the root system of type `S`
+"""
+number_of_roots(S::Symbol, n::Int64) = number_of_roots(RootSystem(S, n))
+
+@doc raw"""
+  getindex(R::RootSystem, r::Int)
+
+Return the `r`-th root of the root system `R`.
+"""
+getindex(R::RootSystem, r::Int) = getindex(R.roots, r)
+
+@doc raw"""
+  root_system_type(R::RootSystem)
+
+Return the Dynkin type of the root system `R`.
+"""
+root_system_type(R::RootSystem) = R.root_system_type
+
+###############################################################################
+#
+#   String I/O
+#
+###############################################################################
+
+function show(io::IO, R::RootSystem)
+  print(io, "Root system of type ")
+  show(io, string(R.root_system_type[1]) * string(R.root_system_type[2]))
+end
+
+###############################################################################
+#
+#   Comparison
+#
+###############################################################################
+
+function Base.:(==)(R1::RootSystem, R2::RootSystem)
+  return R1.roots == R2.roots &&
+         R1.root_system_type == R2.root_system_type
+end
+
+function Base.hash(R::RootSystem, h::UInt)
+  b = 0x9d96557cb5f07773 % UInt
+  h = hash(R.positive_roots, h)
+  h = hash(R.root_system_type, h)
+  h = hash(R.roots, h)
+  h = hash(R.simple_roots, h)
+  return xor(h, b)
+end
+###############################################################################
+#
+#   Constructor
+#
+###############################################################################
+@doc raw"""
+  root_system(S::String) -> root_system
+Return the root system of type `S` where `S` is a string consisting out of
+a letter `A`, `B`, `C`, `D`, `E`, `F`, `G` followed by an integer.
+For the exceptional root system, the integers are fixed, e.g. `G2`, `F4`, `E6`, ...
+"""
+function root_system(S::Symbol, n::Int64)
+  @req S in [:A, :B, :C, :D, :E, :F, :G] "Unknown Dynkin type"
+  # S is a symbol detailing the type of the indecomposable root system 
+  # e.g. "A", "B", "C",... and n is an integer for the number of simple roots
+  return RootSystem(S, n)
+end
+
+###############################################################################
+#
+#   further functions
+#
+###############################################################################
+ @doc raw"""
+   cartan_matrix(S::String) -> Matrix{QQFieldElem}
+
+ Return the Cartan matrix of the type of root system specified by `S`
+  """
+function cartan_matrix(S::Symbol, n::Int64)
+  @req S in [:A, :B, :C, :D, :E, :F, :G] "Unknown Dynkin type"
+  S1 = GAP.Obj(S)
+  RS = GAP.Globals.RootSystem(S1, n)
+  CG = GAP.Globals.CartanMatrix(RS)
+  C = matrix(QQ, CG)
+  return C
+end
```

Comment:
This would then simplify to
```suggestion
function cartan_matrix(S::Symbol, n::Int64)
  return cartan_matrix(root_system(S, n))
end
```